runtime/crab/src/migration.rs

```diff
@@ -41,7 +75,48 @@ impl frame_support::traits::OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 }
 
 fn migrate() -> frame_support::weights::Weight {
-	frame_support::weights::Weight::zero()
+	#[derive(codec::Encode, codec::Decode, sp_runtime::RuntimeDebug)]
+	struct VestingInfo {
+		locked: Balance,
+		per_block: Balance,
+		starting_block: BlockNumber,
+	}
+
+	if let Ok(k) = array_bytes::hex2bytes(BROKEN_STORAGES[0]) {
+		// If `is_some` which means it hasn't been migrated yet.
+		// But actually, without this correction, the account migration will fail.
+		if unhashed::get::<VestingInfo>(&k).is_some() {
+			log::info!("purge `storage({})`", BROKEN_STORAGES[0]);
+			unhashed::kill(&k);
+		}
+	}
+	if let Ok(k) = array_bytes::hex2bytes(BROKEN_STORAGES[1]) {
+		// If `is_some` which means it hasn't been migrated yet.
+		// But actually, without this correction, the account migration will fail.
+		if unhashed::get::<VestingInfo>(&k).is_some() {
+			let v = vec![VestingInfo {
+				locked: (100_000_000_000_u128
+					- (CRAB_LAST_FINALIZE_HEIGHT - 8_421_033) as Balance * 1_u128)
+					* DECIMAL_OFFSET,
+				// Crab2's block time is twice longer than Crab.
+				per_block: 1 * DECIMAL_OFFSET * 2,
+				starting_block: 0,
```

Comment:
As the crab 2.0 has launched, so I think the `starting_block` should be the migration execute number rather than 0.

Reply:
Nope. I use the genesis state.

Reply:
Okay. I got it. But in this way, you need to update the `locked` field?  take the crab 2.0 current block number into account.

```
100_000_000_000_u128 - (CRAB_LAST_FINALIZE_HEIGHT - 8_421_033) * 1 - crab_2.0 number * 2
```

Reply:
Nope. Genesis state doesn't care about the future block number.